x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx

```diff
@@ -0,0 +1,274 @@
+          await setInferenceModels();
+        } else {
+          setInferenceAddError(error.message);
+        }
+      } catch (exception) {
+        setInferenceAddError(exception.message);
+      }
+    },
+    [isInferenceFlyoutVisible]
+  );
+  useEffect(() => {
+    const subscription = subscribe((updateData) => {
+      const formData = updateData.data.internal;
+      const value = formData.main;
+      onChange(value);
+    });
+
+    return subscription.unsubscribe;
+  }, [subscribe, onChange]);
+  const selectedOptions = options.filter((option) => option.checked).find((k) => k.label);
+  const [isInferencePopoverVisible, setIsInferencePopoverVisible] = useState<boolean>(false);
+  const inferencePopover = () => {
+    return (
+      <EuiPopover
+        button={
+          <>
+            <UseField path="main" config={fieldConfigModelId}>
+              {(field) => (
+                <>
+                  <EuiText size="xs">
+                    <p>
+                      <strong>{field.label}</strong>
+                    </p>
+                  </EuiText>
+                  <EuiSpacer size="xs" />
+                  <EuiButton
+                    iconType="arrowDown"
+                    iconSide="right"
+                    color="text"
+                    onClick={() => {
+                      setIsInferencePopoverVisible(!isInferencePopoverVisible);
+                    }}
+                  >
+                    <FormattedMessage
+                      id="xpack.idxMgmt.mappingsEditor.parameters.inferenceId.popover.button"
+                      defaultMessage="{defaultValue}"
+                      values={{ defaultValue: selectedOptions?.label }}
+                    />
+                  </EuiButton>
+                </>
+              )}
+            </UseField>
+          </>
+        }
+        isOpen={isInferencePopoverVisible}
+        panelPaddingSize="m"
+        closePopover={() => setIsInferencePopoverVisible(!isInferencePopoverVisible)}
+      >
+        <EuiContextMenuPanel>
+          <EuiContextMenuItem
+            key="addInferenceEndpoint"
+            icon="plusInCircle"
+            size="s"
+            data-test-subj="addInferenceEndpointButton"
+            onClick={() => {
+              setIsInferenceFlyoutVisible(!isInferenceFlyoutVisible);
+              setIsInferencePopoverVisible(!isInferencePopoverVisible);
+            }}
+          >
+            {i18n.translate(
+              'xpack.idxMgmt.mappingsEditor.parameters.inferenceId.popover.addInferenceEndpointButton',
+              {
+                defaultMessage: 'Add inference Endpoint',
+              }
+            )}
+          </EuiContextMenuItem>
+          <EuiHorizontalRule margin="none" />
+          <EuiContextMenuItem
+            key="manageInferenceEndpointButton"
+            icon="gear"
+            size="s"
+            data-test-subj="manageInferenceEndpointButton"
+            onClick={() => {}}
+          >
```

Comment:
@mdefazio also where should management inference endpoint  go, I suppose trained model page?